cli/src/pcluster/resources/imagebuilder/update_and_reboot.yaml

```diff
@@ -149,13 +256,47 @@ phases:
                 else
                   package-cleanup -y --oldkernels --count=1
                 fi
+                # Install kernel-devel during OS update, so that headers are aligned with new kernel. 
+                # The same is done for Debian through `apt-get -y install linux-aws`
                 yum -y install kernel-devel
+
               elif [[ ${!PLATFORM} == DEBIAN ]]; then
-                while [ ! -f /var/lib/cloud/instance/boot-finished ]; do echo 'Waiting for cloud-init...'; sleep 1; done
-                flock $(apt-config shell StateDir Dir::State/d | sed -r "s/.*'(.*)\/?'$/\1/")/daily_lock systemctl disable --now apt-daily.timer apt-daily.service apt-daily-upgrade.timer apt-daily-upgrade.service
-                sed "/Update-Package-Lists/s/\"1\"/\"0\"/; /Unattended-Upgrade/s/\"1\"/\"0\"/;" /etc/apt/apt.conf.d/20auto-upgrades > "/etc/apt/apt.conf.d/51pcluster-unattended-upgrades"
-                DEBIAN_FRONTEND=noninteractive apt-get -y update && DEBIAN_FRONTEND=noninteractive apt-get -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" --with-new-pkgs upgrade && apt-get --purge autoremove -y
-                apt-get -y install linux-aws linux-headers-aws linux-image-aws
+                DEBIAN_FRONTEND=noninteractive apt-get -y update
+                DEBIAN_FRONTEND=noninteractive apt-get -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" --with-new-pkgs upgrade
+                apt-get --purge autoremove -y
+
+                if [[ ${!DISABLE_KERNEL_UPDATE} != true ]]; then
+                   # already installed to LTS version
+                   apt-get -y install linux-aws linux-headers-aws linux-image-aws
```

Comment:
Why is this necessary? Wouldn't it be picked up by the previous upgrade? In case `$DISABLE_KERNEL_UPDATE != true` we shouldn't have removed them from the base AMI, should we?

Reply:
This code was already there. I imagine the reason is that these packages (in particular linux-aws) are not always installed (expecially in case of custom AMIs). With this code we're sure we have them.